proxy/src/main/java/net/md_5/bungee/util/ChatComponentTransformer.java

```diff
@@ -32,6 +34,32 @@
      */
     private static final Pattern SELECTOR_PATTERN = Pattern.compile( "^@([pares])(?:\\[([^ ]*)\\])?$" );
 
+    public BaseComponent[] legacyHoverTransform(ProxiedPlayer player, BaseComponent... components)
+    {
+        if ( player.getPendingConnection().getVersion() <= ProtocolConstants.MINECRAFT_1_16 )
+        {
+            for ( int i = 0; i < components.length; i++ )
+            {
+                BaseComponent next = components[ i ];
+                if ( next.getHoverEvent().isLegacy() )
+                {
+                    continue;
+                }
+                next = next.duplicate();
+                next.getHoverEvent().setLegacy( true );
+                if ( next.getHoverEvent().getContents().size() > 1 )
+                {
+                    HoverEvent.Content exception = next.getHoverEvent().getContents().get( 0 );
```

Comment:
Instead of doing this, we could just add the other Text Contents as extra? Thoughts?